hail/python/hail/expr/functions.py

```diff
@@ -3236,9 +3236,94 @@ def reversed(x):
     return x
 
 
+@typecheck(name=builtins.str,
+           exprs=tupleof(Expression),
+           filter_missing=builtins.bool,
+           filter_nan=builtins.bool)
+def _comparison_func(name, exprs, filter_missing, filter_nan):
+    if builtins.len(exprs) < 1:
+        raise ValueError(f"{name:!r} requires at least one argument")
+    if (builtins.len(exprs) == 1
+            and (isinstance(exprs[0].dtype, (tarray, tset)))
+            and is_numeric(exprs[0].dtype.element_type)):
+        [e] = exprs
+        if filter_nan and e.dtype.element_type in (tfloat32, tfloat64):
+            name = 'nan' + name
+        return array(e)._filter_missing_method(filter_missing, name, exprs[0].dtype.element_type)
+    else:
+        if not builtins.all(is_numeric(e.dtype) for e in exprs):
+            expr_types = ', '.join("'{}'".format(e.dtype) for e in exprs)
+            raise TypeError(f"{name!r} expects a single numeric array expression or multiple numeric expressions\n"
+                            f"  Found {builtins.len(exprs)} arguments with types {expr_types}")
+        unified_typ = unify_types_limited(*(e.dtype for e in exprs))
+        ec = coercer_from_dtype(unified_typ)
+        indices, aggs = unify_all(*exprs)
+
+        func_name = name
+        if filter_missing:
+            func_name += '_ignore_missing'
+        if filter_nan and unified_typ in (tfloat32, tfloat64):
+            func_name = 'nan' + func_name
+        return construct_expr(functools.reduce(lambda l, r: Apply(func_name, l, r), [ec.coerce(e)._ir for e in exprs]),
+                              unified_typ,
+                              indices,
+                              aggs)
+
+
 @typecheck(exprs=expr_oneof(expr_numeric, expr_set(expr_numeric), expr_array(expr_numeric)),
-           filter_missing=bool)
-def max(*exprs, filter_missing: bool = True) -> NumericExpression:
+           filter_missing=builtins.bool)
+def nanmax(*exprs, filter_missing: builtins.bool = True) -> NumericExpression:
+    """Returns the maximum value of a collection or of given arguments, excluding NaN.
+
+    Examples
+    --------
+
+    Compute the maximum value of an array:
+
+    >>> hl.eval(hl.max([1, 3, 5, 6, 7, 9]))
+    9
+
+    Compute the maximum value of arguments:
```

Comment:
maybe ought to preserve the previous word choice, "take the (minimum,maximum) value"